vizro-core/examples/default/app.py

```diff
@@ -8,21 +8,20 @@
 from vizro.actions import export_data, filter_interaction
 from vizro.tables import dash_data_table
 
+df = px.data.gapminder()
+df_mean = (
+    df.groupby(by=["continent", "year"]).agg({"lifeExp": "mean", "pop": "mean", "gdpPercap": "mean"}).reset_index()
+)
 
-def retrieve_avg_continent_data():
-    """This is function returns gapminder data grouped by continent."""
-    df = px.data.gapminder()
-    mean = (
-        df.groupby(by=["continent", "year"]).agg({"lifeExp": "mean", "pop": "mean", "gdpPercap": "mean"}).reset_index()
-    )
-    return mean
+df_transformed = df.copy()
+df_transformed["lifeExp"] = df.groupby(by=["continent", "year"])["lifeExp"].transform("mean")
+df_transformed["gdpPercap"] = df.groupby(by=["continent", "year"])["gdpPercap"].transform("mean")
+df_transformed["pop"] = df.groupby(by=["continent", "year"])["pop"].transform("sum")
+df_concat = pd.concat([df_transformed.assign(color="Continent Avg."), df.assign(color="Country")], ignore_index=True)
```

Comment:
I have only just read this code properly for the first time. I think in its current form it is actually largely confusing: We are associating the mean and sum across continents to individual countries again without changing the column names to reflect this, or without eliminating the countries so that we only look a continent and year.

I would probably change this to something like:

```py
df_transformed = df.groupby(by=["continent", "year"]).agg({"lifeExp": "mean", "pop": "sum", "gdpPercap": "mean"}).reset_index()
```

Reply:
I think this may be related to the reason why the average shows up weirdly (including being labeled with Zimbabwe here): 
<img width="532" alt="image" src="https://github.com/mckinsey/vizro/assets/83698606/fe4a1d44-44ec-4a74-ba53-cc890a92ae3d">

Can we check this quickly?

Reply:
I tried your suggestion above before actually, but it's not the same 😄 If you do above, it will only give you the country results - not the country next to the continent average result.

Reply:
The results also seem coherent for me? 
![Screenshot 2023-12-13 at 16 28 35](https://github.com/mckinsey/vizro/assets/90609403/6be24143-964e-48f8-b85b-ac713e158a94)

The chart on the right only makes sense if you filter on one country, so select a cell in the table

Reply:
Oh haha ok :)

Hm ok that makes more sense. I guess it's not quite ideal, but I can see why now!

Reply:
Is it good to go for you then? 😄 Would merge after your approval